### ISSUE TITLE
Background task hanging. Most editor features broken. High CPU usage.

### ISSUE THREAD

**Issue by N8th8n8el** [zed-industries/zed#4988]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

I don't know what cause this, but the symptoms were
- syntax highlighting stopped working (didn't update on edits)
- Couldn't save the file (just did nothing, the blue dot in the file tab remained)
- Command panel search didn't work anymore
- 100% CPU usage on several cores
- I could still switch tabs, edit buffers, etc...

### Environment

Zed: v0.74.3 (stable)
OS: macOS 13.1.0
Memory: 16 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

![Screenshot 2023-02-26 at 19 23 01](https://user-images.githubusercontent.com/19362696/221429424-0fce974f-2b47-4734-90aa-fc09b534629c.png)
![Screenshot 2023-02-26 at 19 22 44](https://user-images.githubusercontent.com/19362696/221429430-10d0ca59-1703-45dc-952b-1afeb69dbd4c.png)

https://user-images.githubusercontent.com/19362696/221429437-2b087d65-ab70-4edd-b009-5a7f27e7b71b.mov

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

This is the only interesting line (likely from when I quit Zed):

```
2023-02-26T18:23:24 [ERROR] timed out waiting on app_will_quit
```

The next line is 
```
2023-02-26T18:23:27 [INFO] ========== starting zed ==========
```
the previous ~1000 lines are 
```
2023-02-26T18:19:03 [INFO] completion out of expected range
```
(likely unrelated)

**Comment by nathanael-ruf** [comment]
Haven't had this for over a year. I think we can close this @maxbrunsfeld.

**Comment by JosephTLyons** [comment]
I think we will close this out, since a lot has improved with Zed since this issue was created, but feel free to make a new issue if you run into it again.
